frontend/src/sections/Hero/Hero.tsx

```diff
@@ -1,11 +1,21 @@
 'use client'
 
 import Image from 'next/image'
+import { useState } from 'react'
+import Update from '../Latestupdate/LatestUpdate'
+import Modal from '@/components/Modal/modal'
 
 const HeroSection = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const toggleModal = () => {
+    setIsModalOpen(!isModalOpen)
```

Comment:
```suggestion
    setIsModalOpen((prevState) => !prevState);
```

This is better

Reply:
I find lesser lines of code is better. No ?

Reply:
can change to `const toggleModal = () => setIsModalOpen((prevState) => !prevState);`

Reply:
The current approach might create issues when states are updating asynchronously. Its always better to update state with taking into account its previous value in such use cases